app/basic/page.tsx

```diff
@@ -1,14 +1,54 @@
-export default function Basic() {
+import fs from 'fs';
+import Link from 'next/link';
+import matter from 'gray-matter';
+import { CardMetadata } from '@/types';
+
+const getCardMetadata = (): CardMetadata[] => {
+  const folder = 'content/';
+  const files = fs.readdirSync(folder);
+  const mdxCards = files.filter((file) => file.endsWith('.mdx'));
+  // const slugs = mdxCards.map((file) => file.replace('.mdx', ''));
+  // return slugs;
+
+  const cards = mdxCards.map((fileName) => {
+    const fileContents = fs.readFileSync(`content/${fileName}`, 'utf-8');
+    const matterResult = matter(fileContents);
+
+    return {
+      title: matterResult.data.title,
+      content: matterResult.data.content,
+      slug: fileName.replace('.mdx', ''),
+    };
+  });
+
+  return cards;
+};
+
+const Basic = () => {
+  const cardMetadata = getCardMetadata();
+  const cardPreview = cardMetadata.map((card, title) => (
+    <div className='flex gap-6 mt-6' key={title}>
+      <div className='p-8 rounded-md shadow-md bg-white'>
+        <Link href={`/content/${card.slug}`}>
+          <h3 className='text-xl font-semibold'>{card.title}</h3>
+        </Link>
+      </div>
+    </div>
+  ));
```

Comment:
Everything is fine here, but just wondering about the reasoning for having it in a separate variable and not placing `.map` directly inside return